ksql-api/src/main/java/io/confluent/ksql/api/server/BufferedPublisher.java

```diff
@@ -125,12 +125,15 @@ public boolean accept(final T t) {
   /**
    * If you set a drain handler. It will be called if, after delivery is attempted there are zero
    * elements buffered internally and there is demand from the subscriber for more elements. Drain
-   * handlers are one shot handlers, after being it will never be called more than once.
+   * handlers are one shot handlers, after being called it will never be called more than once.
    *
    * @param handler The handler
    */
   public void drainHandler(final Runnable handler) {
     checkContext();
+    if (drainHandler != null) {
+      throw new IllegalStateException("drainHandler already set");
+    }
```

Comment:
nit: validate params that will be stored in object state; ensuring object does not get into an invalid state, i.e. null check `handler`.